pkg/cosign/env/env.go

```diff
@@ -0,0 +1,120 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package env
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+)
+
+type Variable string
+
+type VariableOpts struct {
+	Description string
+	Sensitive   bool
+}
+
+func (v Variable) String() string {
+	return string(v)
+}
+
+const (
+	VariableExperimental     Variable = "COSIGN_EXPERIMENTAL"
+	VariableDockerMediaTypes Variable = "COSIGN_DOCKER_MEDIA_TYPES"
+	VariablePassword         Variable = "COSIGN_PASSWORD"
+	VariablePKCS11Pin        Variable = "COSIGN_PKCS11_PIN"
+	VariablePKCS11ModulePath Variable = "COSIGN_PKCS11_MODULE_PATH"
+	VariableRepository       Variable = "COSIGN_REPOSITORY"
+)
+
+var (
+	environmentVariables = map[Variable]VariableOpts{
+		VariableExperimental: {
+			Description: "enables experimental cosign features",
+			Sensitive:   false,
+		},
+		VariableDockerMediaTypes: {
+			Description: "to be used with registries that do not support OCI media types",
+			Sensitive:   false,
+		},
+		VariablePassword: {
+			Description: "overrides password inputs with this value",
+			Sensitive:   true,
+		},
+		VariablePKCS11Pin: {
+			Description: "to be used if PKCS11 PIN is not provided",
+			Sensitive:   true,
+		},
+		VariablePKCS11ModulePath: {
+			Description: "is PKCS11 module-path",
+			Sensitive:   false,
+		},
+		VariableRepository: {
+			Description: "can be used to store signatures in an alternate location",
+			Sensitive:   false,
+		},
+	}
+)
+
+func Getenv(name Variable) string {
+	return os.Getenv(name.String())
+}
+
+func LookupEnv(name Variable) (string, bool) {
+	return os.LookupEnv(name.String())
+}
+
+func PrintEnv(showDescription, showSensitive bool) {
```

Comment:
Perhaps this should be moved to `cli/`? WDYT?

Reply:
The reason why I didn't want this in `cli/` is to avoid publicly exposing `environmentVariables` slice. This is because I want to avoid someone changing the slice outside the env package. If this is important, I can workaround it with some getter or I can make the slice public.

Reply:
Good thought. I think it's prudent to avoid letting callers mutate that slice.

I think it's worth it to add a getter, though.

Reply:
This is now moved to `cli/`.